### PR TITLE
Use tokio in tests.

### PR DESCRIPTION
**Pull request by twittner**

### CODE REVIEW

**Comment by mxinden** [comment]
Out of curiosity, could you expand on the motivation for this patch?

**Comment by twittner** [comment]
Like in https://github.com/libp2p/rust-libp2p/issues/1588, tests do not complete with async-std > 1.5.0.

File: src/frame/io.rs [hunk 1]
```diff
@@ -232,7 +232,7 @@ mod tests {
     #[test]
     fn encode_decode_identity() {
         fn property(f: Frame<()>) -> bool {
-            async_std::task::block_on(async move {
+            futures::executor::block_on(async move {
```
Comment (mxinden) [inline src/frame/io.rs:235]: Why use `futures::executor` here and `Runtime::block_on` in other places?

Comment (twittner) [reply src/frame/io.rs:235]: The correctness does not depend on the runtime choice so it does not really matter. As this is a small property test without concurrency, executing it on the same thread is more light-weight.

**Comment by twittner** [comment]
NB: CI [is failing](https://travis-ci.org/github/paritytech/yamux/jobs/697587466) with nightly because of https://github.com/rust-lang/cargo/issues/8351:

```
$ cargo build --verbose
    Updating crates.io index
error: failed to download `futures v0.3.5`
Caused by:
  unable to get packages from source
Caused by:
  failed to parse manifest at `/home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.3.5/Cargo.toml`
Caused by:
  readme file with name '../README.md' was not found
```

**Pull request merged**